<H1>ACTIVTIY1: PALMER PENGUINS DATA ANALYTICS</H1>
<HR>
<H3>ANALYST: MARIA ASHLEY JOY ENALLO</H3>

In [1]:
import numpy as np
import pandas as pd
from scipy.stats import f_oneway
import warnings
warnings.filterwarnings('ignore')

In [2]:
penguin = pd.read_csv('dataset\\penguins.csv')

In [3]:
penguin

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h2>QUESTION#1: HOW MANY FEMALE BORN IN 2007?</h2>
<HR>

In [4]:
year_female = penguin.groupby(['sex','year']).size()

In [5]:
year_female

sex     year
female  2007    51
        2008    56
        2009    58
male    2007    52
        2008    57
        2009    59
dtype: int64

<H2>INSIGHT: BASED ON THE DATA ABOVE THERE ARE 51 FEMALE PENGUIN BORN IN 2007</h2>
<HR>

<h2>QUESTION#2: HOW MANY MALE BORN IN DREAM ISLAND?</h2>
<HR>

In [6]:
male_dream = penguin.groupby(['island', 'sex']).size()

In [7]:
male_dream

island     sex   
Biscoe     female    80
           male      83
Dream      female    61
           male      62
Torgersen  female    24
           male      23
dtype: int64

<h2>INSIGHT: THERE ARE 62 MALE BORN/LIVE IN DREAM ISLAND</h2>
<HR>

<h2>QUESTION#3: IS THERE A CORRELATION IN PENGUINS FLIPPER AND BODY MASS?</h2>
<HR>

In [8]:
flipper_mass = penguin['flipper_length_mm'].corr(penguin['body_mass_g'])

In [9]:
flipper_mass

np.float64(0.8712017673060116)

<h2>INSIGHT: THEREFORE, THERE IS A HIGH CORRELATION BETWEEN PENGUINS FLIPPER AND BODY MASS</h2>
<HR>

<h2>QUESTION#4: AVERAGE BILL_LENGTH_MM MALES</h2>
<HR>

In [10]:
penguin[penguin['sex'] == 'male']['bill_length_mm'].mean()

np.float64(45.85476190476191)

<h2>INSIGHT: THERE IS AN AVERAGE OF 45.85476190476191 IN MALES BILL LENGTH</h2>
<HR>

<h2>QUESTION#5: WHAT SPECIES COMMONLY FOUND IN TORGERSEN?</h2>
<HR>

In [11]:
penguin[penguin['island'] == 'Torgersen']['species'].value_counts()

species
Adelie    52
Name: count, dtype: int64

<h2>INSIGHT: THE SPECIES COMMONLY FOUND IN TORGERSEN IS ADELIE</h2>
<HR>

<h2>QUESTION#6: HOW MANY PENGUINS BORN/ALIVE IN YEAR 2009?</h2>
<HR>

In [12]:
year = penguin.groupby('year').size()

In [13]:
year

year
2007    110
2008    114
2009    120
dtype: int64

<h2>INSIGHT: 120 PENGUINS BORN OR ALIVE IN YEAR 2009</h2>
<HR>

<h2>QUESTION#7: AVERAGE BODY MASS IN BISCOE</h2>
<HR>

In [14]:
penguin[penguin['island'] == 'Biscoe']['body_mass_g'].mean()

np.float64(4716.017964071856)

<h2>INSIGHT: THE AVERAGE MASS OF PENGUINS IN BISCOE IS 4716.017964071856</h2>
<HR>

<h2>QUESTION#8: HOW MANY FEMALE PENGUINS LIVE IN EACH ISLAND?</h2>
<HR>

In [15]:
penguin[penguin['sex'] == 'female']['island'].value_counts()

island
Biscoe       80
Dream        61
Torgersen    24
Name: count, dtype: int64

<h3>INSIGHT: THEREFORE, 80 FEMALE PENGUINS LIVE IN BISCOE, 61 IN DREAM AND 24 IN TORGERSEN</h3>
<hr>

<h3>QUESTION#9: HOW MANY MALE PNGUINS LIVE IN EACH ISLAND?</h3>
<hr>

In [16]:
penguin[penguin['sex'] == 'male']['island'].value_counts()

island
Biscoe       83
Dream        62
Torgersen    23
Name: count, dtype: int64

<h3>INSIGHT: THEREFORE, 83 MALE PENGUINS LIVE IN BISCOE, 62 IN DREAM AND 23 IN TORGERSEN</h3>
<hr>

<h3>QUESTION#10: WHAT GENDER OF PENGUIN COMMONLY FOUND IN EACH ISLAND?</h3>
<hr>

In [17]:
penguin.groupby(['island', 'sex']).size()

island     sex   
Biscoe     female    80
           male      83
Dream      female    61
           male      62
Torgersen  female    24
           male      23
dtype: int64

<h3>INSIGHT: THEREFORE, IN BISCOE MALE IS MORE COMMONLY FOUND AS WELL AS DREAM WHILE IN TORGERSEN FEMALE IS COMMONLY FOUND THE DIFFERENCE IS JUST 1-3 PENGUIN</h3>
<hr>

<h3>QUESTION#11: Which sex (male or female) tends to have the greater body mass on average?</h3>
<hr>

In [18]:
penguin[penguin['sex'] == 'female']['body_mass_g'].mean()

np.float64(3862.2727272727275)

In [19]:
penguin[penguin['sex'] == 'male']['body_mass_g'].mean()

np.float64(4545.684523809524)

<h3>INSIGHT: MALE TEND TO HAVE GREATER BODY MASS BASED ON THEIR AVERAGE </h3>
<hr>

<h3>QUESTION#12: How many penguins in the dataset have a body mass greater than 5,000 grams?</h3>
<hr>

In [20]:
count = (penguin['body_mass_g'] > 5000).sum()

In [21]:
count

np.int64(61)

<h3>INSIGHT: THERE ARE 61 PENGUINS THAT HAS A BODY MASS GREATER THAN 5000 GRAMS </h3>
<hr>

<h3>QUESTION#13: What is the correlation between bill length and bill depth?</h3>
<hr>

In [22]:
length_depth = penguin['bill_length_mm'].corr(penguin['bill_depth_mm'])
length_depth

np.float64(-0.23505287035553274)

<h3>INSIGHT: THERE IS NO CORRELATION BETWEEN BILL LENGTH AND BILL DEPTH</h3>

<h3>QUESTION#14: DOES BODY MASS CORRELATE WITH YEAR OF OBSERVATION(2007,2008,2009)?</h3>
<hr>

In [37]:
year = penguin['year'].corr(penguin['body_mass_g'])
year

np.float64(0.04220939154335576)

<h3>INSIGHT: THERE IS NO CORRELATION BETWEEN YEAR AND BODY MASS OF PENGUINS</h3>
<hr>

<H3>QUESTION#14: What is the standard deviation of body mass for each sex?</H3>
<HR>

In [38]:
sex_bm = penguin[penguin['sex'] == 'male']['body_mass_g'].std()
sex_bm

np.float64(787.6288841581745)

In [39]:
sex_bf = penguin[penguin['sex'] == 'female']['body_mass_g'].std()
sex_bf

np.float64(666.1720495161449)

<h3>INSIGHT: THEREFORE THE STANDARD DEVIATION OF BODY MASS FOR EACH SEX OF PENGUIN IS MALE = 787.6288841581745 FEMALE = 666.1720495161449  </h3>

<H3>QUESTION#15: How many penguins have a flipper length greater than 210 mm?</H3>
<HR>

In [43]:
flipper_p = (penguin['flipper_length_mm'] > 210).sum()
flipper_p

np.int64(100)

<h3>INSIGHT: THERE'S 100 PENGUIN THAT HAS GREATER THAN 210MM FLIPPER LENGTH</h3>

<H3>QUESTION#16: How many penguins have a flipper length greater than 210 mm?</H3>
<HR>

In [46]:
filtered = penguin[penguin["flipper_length_mm"] > 220]
filtered

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
153    154  Gentoo  Biscoe            50.0           16.3              230.0   
169    170  Gentoo  Biscoe            49.2           15.2              221.0   
171    172  Gentoo  Biscoe            48.7           15.1              222.0   
182    183  Gentoo  Biscoe            47.3           15.3              222.0   
185    186  Gentoo  Biscoe            59.6           17.0              230.0   
193    194  Gentoo  Biscoe            49.6           16.0              225.0   
196    197  Gentoo  Biscoe            50.5           15.9              222.0   
199    200  Gentoo  Biscoe            50.5           15.9              225.0   
205    206  Gentoo  Biscoe            50.1           15.0              225.0   
211    212  Gentoo  Biscoe            50.4           15.3              224.0   
213    214  Gentoo  Biscoe            46.2           14.9              221.0   
215    216  Gentoo  Biscoe            54.3           15.7              231.0   
217    218  Gentoo  Biscoe            49.8           16.8              230.0   
219    220  Gentoo  Biscoe            49.5           16.2              229.0   
221    222  Gentoo  Biscoe            50.7           15.0              223.0   
223    224  Gentoo  Biscoe            46.4           15.6              221.0   
224    225  Gentoo  Biscoe            48.2           15.6              221.0   
227    228  Gentoo  Biscoe            48.6           16.0              230.0   
231    232  Gentoo  Biscoe            45.2           16.4              223.0   
233    234  Gentoo  Biscoe            52.5           15.6              221.0   
235    236  Gentoo  Biscoe            50.0           15.9              224.0   
237    238  Gentoo  Biscoe            50.8           17.3              228.0   
241    242  Gentoo  Biscoe            52.1           17.0              230.0   
243    244  Gentoo  Biscoe            52.2           17.1              228.0   
245    246  Gentoo  Biscoe            49.5           16.1              224.0   
247    248  Gentoo  Biscoe            50.8           15.7              226.0   
249    250  Gentoo  Biscoe            46.9           14.6              222.0   
251    252  Gentoo  Biscoe            51.1           16.5              225.0   
253    254  Gentoo  Biscoe            55.9           17.0              228.0   
255    256  Gentoo  Biscoe            49.1           15.0              228.0   
263    264  Gentoo  Biscoe            49.8           15.9              229.0   
265    266  Gentoo  Biscoe            51.5           16.3              230.0   
267    268  Gentoo  Biscoe            55.1           16.0              230.0   
269    270  Gentoo  Biscoe            48.8           16.2              222.0   
273    274  Gentoo  Biscoe            50.4           15.7              222.0   

     body_mass_g     sex  year  
153       5700.0    male  2007  
169       6300.0    male  2007  
171       5350.0    male  2007  
182       5250.0    male  2007  
185       6050.0    male  2007  
193       5700.0    male  2008  
196       5550.0    male  2008  
199       5400.0    male  2008  
205       5000.0    male  2008  
211       5550.0    male  2008  
213       5300.0    male  2008  
215       5650.0    male  2008  
217       5700.0    male  2008  
219       5800.0    male  2008  
221       5550.0    male  2008  
223       5000.0    male  2008  
224       5100.0    male  2008  
227       5800.0    male  2008  
231       5950.0    male  2008  
233       5450.0    male  2009  
235       5350.0    male  2009  
237       5600.0    male  2009  
241       5550.0    male  2009  
243       5400.0    male  2009  
245       5650.0    male  2009  
247       5200.0    male  2009  
249       4875.0  female  2009  
251       5250.0    male  2009  
253       5600.0    male  2009  
255       5500.0    male  2009  
263       5950.0    male  2009  
265       5500.0    male  2009  
267       5850.0    male  2009  
26

In [48]:
species_counts = filtered["species"].value_counts()
species_counts

species
Gentoo    35
Name: count, dtype: int64

<h3>INSIGHT: The most common species with flipper length > 220 mm is species Gentoo with 35 penguins</h3>

<H3>QUESTION#17: What is the maximum flipper length recorded per species?</H3>
<HR>

In [53]:
penguin_clean = penguin.dropna(subset=["flipper_length_mm", "species"])
penguin_clean

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
4        5     Adelie  Torgersen            36.7           19.3   
5        6     Adelie  Torgersen            39.3           20.6   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
4                193.0       3450.0  female  2007  
5                190.0       3650.0    male  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[342 rows x 9 columns]

In [56]:
max_flipper_length = penguin_clean.groupby("species")["flipper_length_mm"].max()
max_flipper_length

species
Adelie       210.0
Chinstrap    212.0
Gentoo       231.0
Name: flipper_length_mm, dtype: float64

<h3>INSIGHT: THE MAXIMUM FLIPPER LENGTH PER EACH SPECIES IS 210 FOR ADELIE, 212 FOR CHINSTRAP, AND 231 FOR GENTOO</h3>